src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginsPageView.axaml.cs

```diff
@@ -16,8 +23,37 @@ public FileOriginsPageView()
                     vm => vm.FileOrigins,
                     v => v.DataGrid.ItemsSource)
                 .DisposeWith(d);
-            
+
+            DataGrid.SelectedItemsToViewModelObservableChangeSetProperty(ViewModel!, vm => vm.SelectedModsObservable)
+                .DisposeWith(d);
+
+            // Synchronize the Grid with DataModel for Selected Items
+            // Enable/Disable Add Mod & Add Mod Advanced Buttons
+            void UpdateAddButtonState(bool isModAdded)
+            {
+                var enable = false;
+                if (!isModAdded)
+                    enable = ViewModel!.SelectedModsCollection.Count > 0 && 
+                             !ViewModel!.SelectedModsCollection.Any(x => x.IsModAddedToLoadout);
+
+                AddModButton.IsEnabled = enable;
+                // Note Add(Advanced) should always be available.
+            }
+
+            ViewModel!.SelectedModsCollection.ObserveCollectionChanges()
+                .Subscribe(_ => { UpdateAddButtonState(false); })
+                .DisposeWith(d);
+            ViewModel!.SelectedModsObservable.WhenValueChanged(model => model.IsModAddedToLoadout)
+                .Subscribe(UpdateAddButtonState)
+                .DisposeWith(d);
```

Comment:
I don't really understand this code. The Add buttons should only be enabled when the user has selected an item, no? This code will do that:

```csharp
            var serialDisposable = new SerialDisposable();
            serialDisposable.DisposeWith(d);

            this.WhenAnyValue(view => view.ViewModel)
                .Select(vm =>
                {
                    if (vm is null) return Disposable.Empty;

                    return vm
                        .WhenAnyObservable(x => x.SelectedModsObservable)
                        .Select(_ => vm.SelectedModsCollection.Count > 0)
                        .SubscribeWithErrorLogging(hasSelection =>
                        {
                            AddModButton.IsEnabled = hasSelection;
                            AddModAdvancedButton.IsEnabled = hasSelection;
                        });
                }).SubscribeWithErrorLogging(disposable =>
                {
                    serialDisposable.Disposable = disposable;
                }).DisposeWith(d);
```

Reply:
Even then, the button enabled state should be controlled by the command `CanExecute` observable, not by manually updating the IsEnabled state

Reply:
Yeah, with command bindings this should be an `canExecute` observable.

Reply:
The behaviour in the proposed code is a bit different.
It only checks for selection. If you delete a mod from the `My Mods` page, the mod could still be re-added.

In any case, I did rewrite it so it's very similar to this, and moved it to the ViewModel, so it uses the `canExecute` on the commands instead of touching the buttons.